Port RP Chems From Cosmic Drift

Eagle0600 opened the pull request:
# Description

As far as RP chems go, all we have right now is "prescribe psiicodine". Nothing else. This aims to add a bit more interest by adding chems targeted at more specific problems.

The chems come in three categories: Antidepressants (serenitol, neurozenium, blissifylovene), anxiety meds (calmafluxine, tranquinase, equilibrazine), and painkillers (stubantazine, soretizone, agonolexyne). Each chem is either strong, medium, or mild, with the risks of each chem increasing with strength. The stronger chems tend to be addictive.

In addition to the stronger RP chems, the following chems are now addictive: Desoxyephedrine, happiness, and impedrezene.

Possible further work:
I didn't want to add or test new systems with this PR, but if I did, I might also consider adding https://github.com/DeltaV-Station/Delta-v/commit/6d7748130361b9be60bc95a7b9041d85ac05e2b2 and https://github.com/DeltaV-Station/Delta-v/pull/2192.
We might also tied antianxiety, antidepressants, and painkillers into our existing mood systems, with or without the above.

---

# TODO

- [x] Cherry-pick https://github.com/DeltaV-Station/Delta-v/pull/1243
- [x] Cherry-pick https://github.com/DeltaV-Station/Delta-v/pull/1248
- [x] Cherry-pick https://github.com/DeltaV-Station/Delta-v/pull/2302
- [x] Check chems are craftable and work as advertised.
- [x] Write description.
- [x] Write changelog.
- [x] Add media.
- [x] Change to the EE mood addiction system.
- [x] Add mood effect locales.
- [x] Possibly add addiction suppression with milder doses.

---

<details><summary><h1>Media</h1></summary>
<p>

![obraz_2024-05-23_170647855](https://private-user-images.githubusercontent.com/96891493/333239274-9b4dbc38-29d3-46fc-91fc-6218f085f4bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzkyNzQtOWI0ZGJjMzgtMjlkMy00NmZjLTkxZmMtNjIxOGYwODVmNGJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkNmE1NjhiNWJhYzRlNzdhYzY2NjYzZDkzYjUzZDkwYjU5MWQ4ZmFkODExZGU2Y2E2MDc4NDQ2ZjUwNTg1NmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7Ot4kBEC1sBWjxRTVxyNowxo5PtJrpV9poRdxVneMoE) ![obraz_2024-05-23_170707424](https://private-user-images.githubusercontent.com/96891493/333239382-cd8d95cf-742c-4675-be18-bc3b28c0444d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzkzODItY2Q4ZDk1Y2YtNzQyYy00Njc1LWJlMTgtYmMzYjI4YzA0NDRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzMTFkNThlY2RkNGU1ZmFjYTVkM2VhNTVkZDM5NGQ5OTU0ZDg2MWVmYzY3NjFkYWZmNWYwYTQ3YWRkYmViYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vq6s0pXdyTr8V9PmOA6ed1XJMbA_fCmcJ-E77wlcsLc) ![obraz_2024-05-23_170711771](https://private-user-images.githubusercontent.com/96891493/333239405-05b902fb-a811-4c75-acd3-49394af8fce5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk0MDUtMDViOTAyZmItYTgxMS00Yzc1LWFjZDMtNDkzOTRhZjhmY2U1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjOWJlYTRjY2Y4YjAwNTAyMjNhZWMzNTVlYTNkNmMzZTBiNTUxOWFiMWNhNGZkMzQ0YTE1MTAzM2U0NTBiYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1UB_opAsHKnkJKAzbW39oK-cVRqRwTh-vMZ70LuBOZg) ![obraz_2024-05-23_170715949](https://private-user-images.githubusercontent.com/96891493/333239432-e254c5c7-0e91-4c38-833e-86c1388344d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk0MzItZTI1NGM1YzctMGU5MS00YzM4LTgzM2UtODZjMTM4ODM0NGQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1ZjAxZTJjZDZmMThjYzgzMmQzMzllMjVmMDMzOWExMGE1MTVmZTcwZDJlNDJkM2VkMWVlMTJjN2Y1NTk0MDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rd8GpRv7B6GnlOc4wXkZz1kS1qUuwUQh4ZIpQx_Ftxs) ![obraz_2024-05-23_170719954](https://private-user-images.githubusercontent.com/96891493/333239457-4c741ce2-1d14-46ef-995e-4e5a3e9474f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk0NTctNGM3NDFjZTItMWQxNC00NmVmLTk5NWUtNGU1YTNlOTQ3NGYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2OGRjNjM4YWYwMTZjZjQyNjExZjA2Nzk2MzAzYTVmNmZhNjJmNzU3NzU2NDYyMjk2MWFlYjc5MDRkNjkxM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pUE2vcQJ0xByW6DFQlYdntI61NpeAVRppaHiJlkMZqY) ![obraz_2024-05-23_170725573](https://private-user-images.githubusercontent.com/96891493/333239510-fe96288a-1a38-4bff-980e-768afe7e651f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk1MTAtZmU5NjI4OGEtMWEzOC00YmZmLTk4MGUtNzY4YWZlN2U2NTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlYzljMGI1YmNiOWJhZmJmM2VjZjI3MzRhN2VlNTJiODc3ODhiODZlMDdjNjVhMmIwZTY5NTRmY2Y2YTI4ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MboZ26VkITBECAKVlcJyCIFIijmdbRiT7K-TzSOkOxE) ![obraz_2024-05-23_170742229](https://private-user-images.githubusercontent.com/96891493/333239617-ec3e9316-5c0d-4fcb-9ace-6c5b6ecb640c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk2MTctZWMzZTkzMTYtNWMwZC00ZmNiLTlhY2UtNmM1YjZlY2I2NDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2MTdhZGE0YWI1MTI5MmVkM2FmZWU4MjFkY2Q0ODY5Yjk5ZTIxNzM1Y2U4NjVhYzllMTkwMGU3Yjg5MzllNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.koybT73daTCuqO8SfqIVW85ee2_ufWAcaj5BHl576dw) ![obraz_2024-05-23_170745825](https://private-user-images.githubusercontent.com/96891493/333239637-f455e2fb-276f-4c63-8049-2e9d993b43ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk2MzctZjQ1NWUyZmItMjc2Zi00YzYzLTgwNDktMmU5ZDk5M2I0M2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlNDA1Y2FjOGQ0YTdlNDEzMTI2NWExNzJiM2M1MDBlMDcyYzUxMGJlNDg0ZDMyNDNlZjBmMTk3ZmI4MzQzOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.auvFo7o0aYZNdtqPkv2OD4i2WbFmvjyDXy1HfFUhT_8) ![obraz_2024-05-23_170749405](https://private-user-images.githubusercontent.com/96891493/333239655-153f9609-7c10-4660-b134-3982e1d343a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MjI0NjEsIm5iZiI6MTc0NTYyMjE2MSwicGF0aCI6Ii85Njg5MTQ5My8zMzMyMzk2NTUtMTUzZjk2MDktN2MxMC00NjYwLWIxMzQtMzk4MmUxZDM0M2E0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDIzMDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MWQxZmE1MmE0YTBhZTU1YmZmOTlkOTZmOTkyZmY0OGIxZGIwMzUzZjRkNjY1Njc2ZWQ0NTM5NDI2NjY1MzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UhoJxdY7vf5eUM8q2DB4l6R3SvFcAztKylnrzgmc314)

</p>
</details>

---

# Changelog

:cl: MilonPL, Stop-Signs, Eagle0600
- add: Added new anxiety meds, antidepressants and painkillers. Psychologists rejoice!
- add: Desoxyephedrine, Happiness, and Impedrezene are now addictive.

MilonPL committed:
port painkillers and antidepressants from cosmatic drift
* add RP chems
* fix??
* make more chems addictive
* makes THC, Space Drugs, Impedrezne and Desoxyephedrine all addictive
* move CD files to the correct folder, add deltav markings, fix guidebook entries

MilonPL committed:
fix addiction popups (#1248)

* fix??

* UGHHH

Stop-Signs committed:
Psych chems no longer stronger than nocturine  (#2302)

Whoever made this was off their rocker

Eagle0600 committed:
Add newlines at end of files.

Eagle0600 committed:
Style reagent names in lower-case

Mnemotechnician commented:
Mmmm, addictine... I have really bad experience with it from Delta-V. It's proven itself to be bad both in theory and in practice. It clogs up your bloodstream and prevents other chems from metabolizing. Different types of "addictives" act as substitutes for each other. It requires adding a lot of boilerplate YML code to different chemicals to function. And more.

I think EE's way of implementing addictions - that is, via the mood system - is far superior.

Mnemotechnician reviewed (changes requested):
This will conflict with the mood system, so something should be done about it before it's merged.

Memeji commented:
> This will conflict with the mood system, so something should be done about it before it's merged.

Yeah I want to get it moved into the mood system as well

Eagle0600 commented:
I'm happy to move the addictions over to the mood system, but I don't know a lot about it.

Eagle0600 commented:
Alright, I'm now looking into chem addictions. Let's say that every addictive new chem from cosmic drift should still be addictive. What existing chems do we want to be addictive? Desoxy, for example, is a very obvious case where it should be. Weed is less clear. Does direction have any comment on this?

Memeji commented:
We can make things like Weed not as addictive.
Because we have more harder drugs that can be addictive, especially with the added ones for RP here.

Eagle0600 commented:
Drugs that the Cosmic Drift port makes addictive: Desoxyephedrine, THC, Impedrezene, SpaceDrugs. Can I get a list from direction that we want to be addictive? At current moment, I'm looking at Deso: Yes, THC: No.

Memeji commented:
> Drugs that the Cosmic Drift port makes addictive: Desoxyephedrine, THC, Impedrezene, SpaceDrugs. Can I get a list from direction that we want to be addictive? At current moment, I'm looking at Deso: Yes, THC: No.

Gimme some time to look into it for you!

Eagle0600 commented:
Thanks. I've already started doing the switchover to moods, but while I completely agree with you that addictine is an awful system, it *does* capture a couple of things that the mood system doesn't: Scaling addiction severity, and difference between gaining/scaling an addiction and suppressing withdrawal. I'm fine skipping over the former, but I'd like a way to do the latter. I think I can add a mood condition to the mood effect such that a lower dosage of a chem can suppress the addiction if and only if you already have the addiction.

Eagle0600 commented:
Oh, and Happiness. It literally says "highly addictive" in the description. It's worth noting that Space Drugs was given a chem mood already, marked as non-addictive, it and ethanol being the only drugs so far to do so, so I'm happy to keep that as-is.

Eagle0600 committed:
Replace addictine with mood-system addiction.
Remove addiction from THC.

Eagle0600 commented:
I've decided for now that impedrezene is addictive, pending direction advice. Testing the first round of changes before I push it.

Eagle0600 committed:
Lower doses can suppress withdrawal symptoms

Eagle0600 commented:
Alright, once more at a place I'm happy enough with it and ready for review.

Eagle0600 commented:
I'm having some errors that I believe are caused by my work. I'll take another look at this another day.

Mnemotechnician commented:
Based on the error message, it seems like you forgot to assign some required field of some object in the Desoxy prototype.

Eagle0600 commented:
Just taking a quick glance at my new mood effect conditions, it seems likely the error is caused by the description field not being nullable. I'll make a quick change and have another go.

Eagle0600 committed:
Fix?

Eagle0600 commented:
That appears to have done it.

Eagle0600 committed:
Improve ChemAddMoodlet guidebook description
Removed stray newline.
Fixed case and tense.
Chems can now say what mood they cause by name.

Eagle0600 committed:
Ammend previous commit.
Revert debug changes not being discarded.
Include mood effect name locales that were left out.

Memeji commented:
I'll make sure to look over the list of chems to be addictive before the next update, so we can get this in for it.

Eagle0600 commented:
> I'll make sure to look over the list of chems to be addictive before the next update, so we can get this in for it.

For now, my current implementation is desoxyephedrine, happiness, and impedrezene. Happiness and impedrezene are moderately addictive, while desoxyephedrine is more highly addictive, but none are guaranteed addiction. I have notes on how likely each drug is to cause an addiction at certain dosages in the reagent prototype comments.

Memeji commented:
> > I'll make sure to look over the list of chems to be addictive before the next update, so we can get this in for it.
> 
> For now, my current implementation is desoxyephedrine, happiness, and impedrezene. Happiness and impedrezene are moderately addictive, while desoxyephedrine is more highly addictive, but none are guaranteed addiction. I have notes on how likely each drug is to cause an addiction at certain dosages in the reagent prototype comments.

Seen! It's great work!

Memeji reviewed (approved):
Looks good from what I can see, nothing comes to the top of my mind for what to add. We can change things as time goes on!
